pkg/blockbuilder/scheduler/queue.go

```diff
@@ -79,15 +87,24 @@ func (q *JobQueue) Enqueue(job *types.Job, priority int) error {
 	jobWithPriority := NewJobWithPriority(job, priority)
 	q.pending.Push(jobWithPriority)
 	q.statusMap[job.ID] = types.JobStatusPending
+	q.cond.Broadcast()
 	return nil
 }
 
 // Dequeue gets the next available job and assigns it to a builder
-func (q *JobQueue) Dequeue(_ string) (*types.Job, bool, error) {
+func (q *JobQueue) Dequeue(ctx context.Context, _ string, wait bool) (*types.Job, bool, error) {
 	q.mu.Lock()
 	defer q.mu.Unlock()
 
-	if q.pending.Len() == 0 {
+	if wait {
```

Comment:
This waits under mutex lock, so it'll likely deadlock the process since nothing can be enqueued during this period.

Reply:
`q.cond.Wait` internally [unlocks the mutex](https://pkg.go.dev/sync#Cond.Wait) while it waits for the signal

Reply:
ah, thanks 👍